### PR TITLE
Fix migration e2es

### PR DESCRIPTION
**Pull request by dpilch**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] E2E test run linked
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/schemas/references.graphql [hunk 1]
```diff
@@ -1,17 +1,17 @@
 type Primary @model @auth(rules: [{ allow: public }]) {
-  id: ID! @primaryKey
```
Comment (dpilch) [inline packages/amplify-e2e-tests/schemas/references.graphql:2]: Just removing this to simplify the test case.

File: packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts [hunk 2]
```diff
@@ -1560,4 +1578,66 @@ const importTable = async (tableDef: CustomDDB.Input): Promise<AWSCDKAsyncCustom
   return result;
 };
 
+export const getImportedTableComparisonProperties = (importedTable: TableDescription): TableComparisonProperties => {
```
Comment (palpatim) [inline packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts:1581]: 1. Does this need to be exported? It looks like it's used above, but I can't tell that it's used elsewhere (maybe it's exported for tests though)
2. This file is already pretty long & unwieldy -- can we refactor the table comparison logic and this new helper into a separate set of files to make it easier to figure out what the heck is going on in here? I'd also suggest you create field-specific helpers for any field that's not a straight up `ComparisonField: importedTable.Field` assignment, again to improve readability. The current structure makes it pretty hard to parse.

Comment (dpilch) [reply packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts:1581]: 1. Exported for tests
2. Would you prefer this in a follow up or done in this PR?

**Review by palpatim — APPROVED** [review]

File: packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/import-table.ts [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+};
+
+const getExpectedStreamSpecification = (createTableInput: CreateTableCommandInput): TableComparisonProperties['StreamSpecification'] => {
+  return createTableInput.StreamSpecification;
+};
+
+const getExpectedProvisionedThroughput = (
+  createTableInput: CreateTableCommandInput,
+): TableComparisonProperties['ProvisionedThroughput'] => {
+  return createTableInput.ProvisionedThroughput || { ReadCapacityUnits: 0, WriteCapacityUnits: 0 };
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/import-table.ts:177]: Any reasoning for the defaulted throughput values here? Do these match the current defaults we use in provisioning?

File: packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/import-table.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        Status: 'ENABLED',
+      }
+    : undefined;
+};
+
+const getExpectedDeletionProtectionEnabled = (
+  createTableInput: CreateTableCommandInput,
+): TableComparisonProperties['DeletionProtectionEnabled'] => {
+  return createTableInput.DeletionProtectionEnabled || false;
+};
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/import-table.ts:193]: similar question about the default choices here.

**Review by phani-srikar — APPROVED** [review]
